### PR TITLE
Rewrite omit

### PR DESCRIPTION
**Pull request by chrisfesler**
**Before**

chrome profile on mac of benchmark showed `omit` is 5.1% of `runMicrotasks`

Benchmark results:

```
  latency: {
    average: 97.1,
    mean: 97.1,
    stddev: 11.75,
    min: 90,
    max: 227,
```

**After**

profile shows `omit` is 0.9% of `runMicrotasks`

```
  latency: {
    average: 90.64,
    mean: 90.64,
    stddev: 6.6,
    min: 84,
    max: 140,
```

### CODE REVIEW

File: packages/textkit/src/run/omit.js [hunk 1]
```diff
@@ -5,11 +5,9 @@
...
  */
 const omit = (value, run) => {
-  const attributes = Object.assign({}, run.attributes);
-
-  delete attributes[value];
-
-  return Object.assign({}, run, { attributes });
+  const newAttributes = {...run.attributes};
+  delete newAttributes[value];
+  return {...run, attributes: newAttributes};
```
Comment (chrisfesler) [inline packages/textkit/src/run/omit.js:10]: So this is a bit faster on my mac.

If this proves to be faster in cloud run, too, then we should consider replacing each of the (many!) uses of `Object.assign()` with the spread operator.

I expect actual performance to vary heavily with the machine where it's running, node version, etc.

File: packages/textkit/src/run/omit.js [hunk 2]
```diff
@@ -5,11 +5,9 @@
  * @return {Object} run without ommited attribute
  */
 const omit = (value, run) => {
```
Comment (pwalker) [inline packages/textkit/src/run/omit.js:7]: Do you know what's different about the compiled versions of these two methods?

Comment (kberezin-nshl) [reply packages/textkit/src/run/omit.js:7]: TL;DR: `assign` modifies the object in place and calls setters vs spread operator creates a shallow copy.
Some interesting performance insights here: 
* https://www.reddit.com/r/node/comments/rp6jb8/discussion_object_spread_much_faster_than/

I _think_ we can safely replace assign with spread because nobody refers to "old" objects hoping that those are modified now?

**Review by realark — APPROVED** [review]

File: packages/textkit/src/run/omit.js (hunk 2 above)
Comment (pwalker) [reply packages/textkit/src/run/omit.js:7]: > I think we can safely replace assign with spread because nobody refers to "old" objects hoping that those are modified now?

Since both versions copy the objects, I think it safe to assume the old ones are discarded.  However, I have see a lot manipulation of objects in place in this repo, so I think we need to tread carefully in general.

> TL;DR: assign modifies the object in place and calls setters vs spread operator creates a shallow copy.
Some interesting performance insights here:

What I was trying to refer to is what does the code look like when it gets transpiled with babel?  I think it could be important to consider when working on micro-optimizations.

Also, maybe it could be worth checking out this codemod? https://github.com/babel/babel/tree/main/codemods/babel-plugin-codemod-object-assign-to-object-spread

Comment (pwalker) [reply packages/textkit/src/run/omit.js:7]: Also, btw, I looked at the compiled versions of both, and they look the same as the source, so I think this is good to go.

**Review by pwalker — APPROVED** [review]
